spec/emerald/rubify_spec.rb

```diff
@@ -0,0 +1,131 @@
+require "spec_helper"
+require "emerald/rubify"
+require "emerald/atom"
+require "emerald/string"
+require "emerald/number"
+require "emerald/list"
+
+RSpec.describe Emerald::Rubify do
+
+  it "compiles an empty source" do
+    compiled_code = Emerald::Rubify.new([]).rubify
+    expect(compiled_code).to eq('')
+  end
+
+  describe "code generation from atoms" do
+    it "generates code from a single atom" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new("hello-world")]).rubify
+      expect(compiled_code).to eq('hello-world')
+    end
+
+    it "generates code from two atoms" do
+      compiled_code = Emerald::Rubify.new([Emerald::Atom.new("hello"),Emerald::Atom.new("world")]).rubify
+      expect(compiled_code).to eq('hello world')
+    end
+  end
+
+  describe "code generation from numbers" do
+    it "generates code from a single number" do
+      compiled_code = Emerald::Rubify.new([Emerald::Number.new(5.0)]).rubify
+      expect(compiled_code).to eq('5.0')
+    end
+
+    it "generates code from two numbers" do
+      compiled_code = Emerald::Rubify.new([Emerald::Number.new(1.0),Emerald::Number.new(3.0)]).rubify
+      expect(compiled_code).to eq('1.0 3.0')
```

Comment:
The numbers would need to either be on separate lines, or they'd need to have a semicolon between them.

```irb
louis ~ $ irb
irb(main):001:0> 1.0 3.0
SyntaxError: (irb):1: syntax error, unexpected tFLOAT, expecting end-of-input
	from /Users/louispilfold/.rbenv/versions/2.3.0/bin/irb:11:in `<main>'
```